src/riscv64/getcontext.S

```diff
@@ -0,0 +1,34 @@
+/* libunwind - a platform-independent unwind library
+   Copyright (C) 2008 CodeSourcery
+        
+This file is part of libunwind.
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
+
+#include "offsets.h"
+#include <endian.h>
+
+        .text
+        .global _Uriscv64_getcontext
+        .type   _Uriscv64_getcontext, %function
+        # This is a stub version of getcontext() for riscv64.
+_Uriscv64_getcontext:
+
+
```

Comment:
Why does this exist? Anything using it will instantly break, there isn't even a ret, you just fall through into whatever code happens to come after it.

Reply:
Whoops, this bit was hastily copied from `tilegx`. It's unused and I'll remove it.

Reply:
Well, the symbol seems to be required by the check-namespace test. Instead of making an exception for riscv64, I'll add a dummy function and indicate that it's unused.

Reply:
Why not implement it properly? It's just a bunch of stores to a pointer.

Reply:
I was under the assumption that it's better to just use glibc's getcontext instead of doing it ourselves. What's the rationale for doing this?

Reply:
On the platforms where getcontext is not available (listed in the README), this stub is used. note that ucontext was deprecated over a decade ago, so it is safe to assume that it will not always be available in the future. (e.g. alpine links to libucontext: https://gitlab.alpinelinux.org/alpine/aports/-/blob/988b5a82863f9a34942c5e89b8813ea037d80c4c/main/libunwind/APKBUILD#L28 and so on..)

Reply:
Oh I see. Will do that in a bit.

Reply:
Updated the PR.